internal/provider/kubernetes/controller.go

```diff
@@ -121,14 +121,14 @@ type resourceMappings struct {
 	// extensionRefFilters is a map of filters managed by an extension.
 	// The key is the namespaced name of the filter and the value is the
 	// unstructured form of the resource.
-	extensionRefFilters map[types.NamespacedName]unstructured.Unstructured
+	extensionRefFilters map[message.NamespacedNameAndType]unstructured.Unstructured
```

Comment:
can we reuse this structure or create a new similar one?

https://github.com/envoyproxy/gateway/blob/02ff9eb749ba7eda1a53dfcbd80fb2b6f46b23fe/internal/provider/kubernetes/helpers.go#L28

Reply:
all fields are string type should be okay

Reply:
The problem with https://github.com/envoyproxy/gateway/blob/02ff9eb749ba7eda1a53dfcbd80fb2b6f46b23fe/internal/provider/kubernetes/helpers.go#L28 is that it doesn't contain the object's group, only the type.

This means that if it is used as the key, then these two different objects will be considered the same:

```yaml
---
apiVersion: some.group/v1
type: Type1
metadata:
   name: some-name
   namespace: default
---

apiVersion: some.other.group/v1
type: Type1
metadata:
   name: some-name
   namespace: default
```

Reply:
so we can create a new structure with `group` here, instead of creating at `internal/message`

Reply:
Done.